app/utils/callback/lava.py

```diff
@@ -222,7 +226,37 @@ def _get_lava_meta(meta, job_data):
             handler(meta, step["metadata"])
 
 
-def _add_test_log(meta, job_log, base_path, suite):
+def _get_dir_path(meta, base_path):
+    """Parse the meta-data from LAVA
+
+    Update the metadata with the storage path of the artifacts.
+    If possible, use the file_server_resource from the metadata.
+
+    :param meta: The boot meta-data.
+    :type meta: dictionary
+    :param base_path: The .
+    :type job_data: dict
+    """
+    file_server_resource = meta.get(models.FILE_SERVER_RESOURCE_KEY)
+    if file_server_resource:
+        dir_path = os.path.join(
+            base_path,
+            file_server_resource,
+            meta[models.LAB_NAME_KEY])
+    else:
+        dir_path = os.path.join(
+            base_path,
+            meta[models.JOB_KEY],
+            meta[models.GIT_BRANCH_KEY],
+            meta[models.KERNEL_KEY],
+            meta[models.ARCHITECTURE_KEY],
+            meta[models.DEFCONFIG_FULL_KEY],
+            meta[models.BUILD_ENVIRONMENT_KEY],
+            meta[models.LAB_NAME_KEY])
+    meta.update({'dir_path': dir_path})
```

Comment:
This would also work: `meta["dir_path"] = dir_path`.

Reply:
Also, all the other meta-data has its keys defined in `app/models/__init__.py`.  It seems worth adding something like this:
```
STORAGE_PATH = "storage-path"
```
or `DIRECTORY_PATH` or whatever makes sense.

Reply:
And please be explicit about any new fields that are mandatory.

Reply:
@khilman not sure I understand what you mean

Reply:
There are several types of data fields, but mainly: the JSON payload found in HTTP requests, and the meta-data found in the LAVA job definition.  The former ones have checklists in `app/models/__init__.py` for each request type, and the `BUILD_ENVIRONMENT_KEY` is being added there so that's all fine.  For the latter ones, there isn't any check of this kind so if some mandatory meta-data is missing it will typically cause a Python exception.  This is something that would need to be improved, to return an HTTP error if the data is invalid and document things better.  I don't think there are any new mandatory meta-data fields as part of this PR though.